toolkit/tools/pkg/imagecustomizerlib/customizeutils.go

```diff
@@ -394,3 +402,70 @@ func addCustomizerRelease(imageChroot *safechroot.Chroot, toolVersion string, bu
 
 	return nil
 }
+
+func handleSELinux(selinuxMode imagecustomizerapi.SELinux, selinuxDisablePackageCheck bool,
+	partitionsCustomized bool, imageChroot *safechroot.Chroot,
+) error {
+	var err error
+
+	// Resolve the default SELinux mode.
+	if selinuxMode == imagecustomizerapi.SELinuxDefault {
+		selinuxMode, err = getCurrentSELinuxMode(imageChroot)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If the partitions were customized, then the grub.cfg file will have been recreated from scratch and therefore
+	// the SELinux args will already be correct and don't need to be updated.
+	if !partitionsCustomized {
+		// Update the SELinux kernel command-line args.
+		err := updateSELinuxCommandLine(selinuxMode, imageChroot)
+		if err != nil {
+			return fmt.Errorf("failed to update SELinux args in grub.cfg:\n%w", err)
+		}
+	}
+
+	if selinuxMode != imagecustomizerapi.SELinuxDisabled {
+		imagerSELinuxMode, err := selinuxModeToImager(selinuxMode)
+		if err != nil {
+			return err
+		}
+
+		if !selinuxDisablePackageCheck {
+			// Check if the selinux-policy package is installed.
+			err = imageChroot.UnsafeRun(func() error {
+				stdout, _, err := shell.Execute("rpm", "-q", configuration.SELinuxPolicyDefault)
+				if err != nil {
+					// Note: RPM's stdout will say somelike like "package selinux-policy is not installed".
+					stdout = strings.TrimSpace(stdout)
+					return fmt.Errorf("SELinux is enabled so the policy package must be installed:\n%s:\n%w", stdout, err)
+				}
```

Comment:
In case RPM fails for another reason, we may be losing useful debugging output if we don't capture `stderr`. In such cases the error message may be also misleading. I'd suggest making the log a bit more robust. One idea:
```suggestion
				stdout, stderr, err := shell.Execute("rpm", "-q", configuration.SELinuxPolicyDefault)
				if err != nil {
					// Note: RPM's stdout will say somelike like "package selinux-policy is not installed".
					stdout = strings.TrimSpace(stdout)
					stderr = strings.TrimSpace(stderr)
					return fmt.Errorf("failed while verifying if an SELinux policy package is installed:\nStdout: %s:\nStderr: %s\nError: %w", stdout, stderr, err)
				}
```

We could also go smarter and separate the "package X is not installed" from other errors but in that case I'd extract that logic to a separate function.

Reply:
moot